thoth/adviser/steps/security_indicators.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# thoth-adviser
+# Copyright(C) 2020 Kevin Postlethwait
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Score based on security indicators aggregated."""
+
+from typing import Any
+from typing import Optional
+from typing import Tuple
+from typing import List
+from typing import Dict
+from typing import TYPE_CHECKING
+import logging
+
+from thoth.python import PackageVersion
+
+from ..enums import RecommendationType
+from ..exceptions import NotAcceptable
+from ..state import State
+from ..step import Step
+
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SecurityIndicatorStep(Step):
+    """A step that scores a state based on security info aggregated."""
+
+    # Weights for Confidence
+    HIGH_CONFIDENCE_WEIGHT = 1
+    MEDIUM_CONFIDENCE_WEIGHT = 0.5
+    LOW_CONFIDENCE_WEIGHT = 0.1
+
+    # Weights for Security
+    HIGH_SEVERITY_WEIGHT = 100
+    MEDIUM_SEVERITY_WEIGHT = 10
+    LOW_SEVERITY_WEIGHT = 1
```

Comment:
Can this be parametrized? It might be a good idea to do so - in that case we can adjust the pipeline configuration (using JSON input to adviser) and observe the behavior of adviser in different cases.

See https://thoth-station.ninja/docs/developers/adviser/unit.html#unit-configuration